### PR TITLE
ref(js): Remove OrganizationActions

### PR DESCRIPTION
**Pull request by evanpurkhiser**

### CODE REVIEW

File: static/app/actionCreators/onboardingTasks.tsx [hunk 1]
```diff
@@ -37,7 +37,7 @@ export function updateOnboardingTask(
     ? organization.onboardingTasks.map(task =>
         task.task === updatedTask.task ? {...task, ...updatedTask} : task
       )
-    : [...organization.onboardingTasks, {...updatedTask, user}];
+    : [...organization.onboardingTasks, {...updatedTask, user} as OnboardingTaskStatus];
```
Comment (evanpurkhiser) [inline static/app/actionCreators/onboardingTasks.tsx:40]: It's not EXACTLY right, but 🤷

File: static/app/stores/organizationStore.tsx [hunk 2]
```diff
@@ -51,13 +39,16 @@ const storeConfig: OrganizationStoreDefinition = {
     this.trigger(this.get());
   },
 
-  onUpdate(updatedOrg, {replace = false} = {}) {
+  onUpdate(updatedOrg: Organization, {replace = false} = {}) {
```
Comment (evanpurkhiser) [inline static/app/stores/organizationStore.tsx:42]: Also _sort of_ wrong, but 🤷

File: static/app/types/onboarding.tsx [hunk 3]
```diff
@@ -93,7 +93,7 @@ export type OnboardingTaskDescriptor =
 export interface OnboardingTaskStatus {
   status: 'skipped' | 'pending' | 'complete';
   task: OnboardingTaskKey;
-  completionSeen?: string;
+  completionSeen?: string | boolean;
```
Comment (priscilawebdev) [inline static/app/types/onboarding.tsx:96]: good catch

File: static/app/stores/organizationStore.tsx [hunk 4]
```diff
@@ -24,22 +20,14 @@ interface OrganizationStoreDefinition extends CommonStoreDefinition<State> {
   get(): State;
   init(): void;
   onFetchOrgError(err: RequestError): void;
-  onUpdate(org: Organization, options?: UpdateOptions): void;
+  onUpdate(org: Organization, options?: {replace: true}): void;
```
Comment (priscilawebdev) [inline static/app/stores/organizationStore.tsx:23]: two `onUpdate`? 👀

Comment (evanpurkhiser) [reply static/app/stores/organizationStore.tsx:23]: Yeah this is an override so when `replace` is true the full organization is required

**Review by priscilawebdev — APPROVED** [review]

**Pull request merged**